src/models/main-model.js

```diff
@@ -0,0 +1,22 @@
+import {Rx} from '@cycle/core';
+
+import makeVisibleIndices$ from './visible-indices-stream';
+
+function model(actions) {
+  let tableHeight$ = Rx.Observable.just(500);
+  let rowHeight$ = Rx.Observable.just(30);
+  let columns$ = Rx.Observable.just(['ID', 'ID * 10', 'Random Number']);
+  let rowCount$ = Rx.Observable.just(10000);
+  let scrollTop$ = actions.userScrolled$.startWith(0);
+  let visibleIndices$ = makeVisibleIndices$(
+    tableHeight$, rowHeight$, rowCount$, scrollTop$
+  );
+  let state$ = Rx.Observable.combineLatest(
+    tableHeight$, rowHeight$, columns$, rowCount$, visibleIndices$,
+    (tableHeight, rowHeight, columns, rowCount, visibleIndices) =>
+      ({tableHeight, rowHeight, columns, rowCount, visibleIndices})
+  );
+  return state$;
+}
+
+export default model;
```

Comment:
`model()` is supposed to encapsulate all the state and all the "logic" for your app. We take `actions` as input. This came from intent. The Subjects were replaced with `Observable.just`. Also learn the `startWith` trick to give a default value to the Observable (replacing BehaviorSubject logic). I'm returning `state$` as the combination of all these state pieces so that the view doesn't need to do this `combineLatest`, it can simply receive a state Observable emitting state objects.